src/main/java/org/reactome/addlinks/fileprocessors/ensembl/EnsemblBiomartFileProcessor.java

```diff
@@ -0,0 +1,115 @@
+package org.reactome.addlinks.fileprocessors.ensembl;
+
+import org.reactome.addlinks.fileprocessors.FileProcessor;
+
+import java.io.*;
+import java.util.*;
+
+public class EnsemblBiomartFileProcessor extends FileProcessor<Map<String, List<String>>> {
+
+    public EnsemblBiomartFileProcessor()
+    {
+        super(null);
+    }
+
+    public EnsemblBiomartFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    // TODO: Add unit tests
+    /**
+     * This class builds a large, rather unwieldy two-layer mapping structure of identifiers corresponding to a specific species.
+     * The first level has 4 types of keys for each species. These keys are 'uniprotToProteins', 'proteinToGenes',
+     * 'proteinToTranscripts', and 'transcriptToMicroarrays'. These describe the type of mapping that this
+     * key is to. For example, the key 'hsapiens_uniprotToProteins' would access a Map of Human UniProt protein identifiers
+     * to a List of Ensembl protein identifiers. Transcript denotes Ensembl transcript identifiers, Gene is Ensembl gene identifiers
+     * and Microarray denotes microarray probe identifiers.
```

Comment:
It can be useful to show the structure (assuming this is accurate):
```
+uniprotToProteins
|     |
|     +----hsapiens_uniprotToProteins
|     |              |
|     |              +---Q123456 => [ENS00001, ENS00002,...]
|     |              +---Q789012 => [ENS00003, ENS00004,...]
|     +----ggallus_uniprotToProteins
|                    |
|                    +---Q345678 => [ENS00005, ENS00006,...]
|                    +---Q912345 => [ENS00007, ENS00008,...]
+proteinToGenes
|
...
```

Reply:
Thats a good idea. Do you think including this large amount of text in the code is appropriate or should I just include it as part of a README

Reply:
Well, _I_ think it's OK to put into the code, as a comment. I've seen it and done it. I wouldn't worry too much about it being so many lines - if it makes the code that much more clear than it's probably worth it.

I don't think the README is an appropriate place, because this diagram is really only relevant to _one_ small piece of the whole application. Putting it in the class that works with it puts it in the place where it is most likely to be needed, and it keeps clutter out of the README.

Reply:
There is precedent for data structure diagrams in code!

https://github.com/reactome/add-links/blob/196a3e93ad05d890c4b3e30d05c3d9019ec46e03/src/main/java/org/reactome/addlinks/fileprocessors/UniprotFileProcessor.java#L31-L58